src/js/components/add-image/add-image.js

```diff
@@ -58,9 +71,14 @@ class AddImage extends MaharaBaseComponent {
         StateStore.dispatch({ type: FILE_ENTRY.EDIT_ENTRY, imageDetails: imageDetails });
         Router.navigate(PAGE_URL.PENDING);
     }
+
     guidGenerator() {
         return (Math.random() + 1).toString(36).substring(2, 12) + (Math.random() + 1).toString(36).substring(2, 12);
     }
+
+    changeFolder(targetFolderName) {
+        this.setState({'targetFolderName': targetFolderName});
+    }
 }
 
 export default AddImage;
```

Comment:
Again, something that might be better to sort more broadly later, but as you've done in selector.js, I think PropTypes would be a good addition to this and other components.